rasa/shared/core/training_data/visualization.py

```diff
@@ -402,20 +400,40 @@ def _add_message_edge(
     is_current: bool,
 ) -> None:
     """Create an edge based on the user message."""
-
     if message:
-        message_key = message.get("intent", {}).get("name", None)
+        message_intent = message.get("intent", {}).get("name", None)
+        # get the entities as \n separated values
+        entities: Optional[List[Dict[Text, Any]]] = message.get("entities")
+        entity_values = []
+        if entities:
+            entity_values = [
+                value
+                for entity in entities
+                for key, value in entity.items()
+                if key == "entity"
+            ]
+        if entity_values:
+            newline = ",\n"
+            entity_values_as_str = f"{newline}({newline.join(entity_values)})"
+        else:
+            entity_values_as_str = ""
+
+        message_key = message_intent + entity_values_as_str
+        message_label = message_key  # message.get("text", None)
+    elif message and message.get("text"):
+
         message_label = message.get("text", None)
+        message_key = message.get("intent", {}).get("name", None)
```

Comment:
I don't think this will ever get evaluated because it's a subset condition of first if branch condition,  unless you specify in L403: `if message and message.get("text") is None`.

Reply:
Yeah, i updated the first condition, so now, we have two conditions,
if message has entities
then first condition
else if message has text
then second condition